includes/User.php

```diff
@@ -2174,7 +2174,9 @@ public function setPassword( $str ) {
 		}
 
 		$this->setInternalPassword( $str );
-
+		if($clearTokens) {
+			wfRunHooks('SetPassword', [$this->getId()]);
```

Comment:
Did we check all methods that bind to `SetPassword` hook? By making this hook optional we can break things.